core/nuxeo-document.js

```diff
@@ -226,32 +226,59 @@ import './nuxeo-resource.js';
       return path;
     }
 
-    appendClientReason(prop) {
-      prop.viewData = `${prop.data}&clientReason=view`;
-      prop.downloadData = `${prop.data}&clientReason=download`;
+    appendClientReason(prop, subProp, param) {
+      prop[subProp === 'url' ? 'viewUrl' : 'viewData'] = `${prop[subProp]}${param.view}`;
```

Comment:
There is no need to introduce unnecessary complexity in code.
therefore instead of managing 2 different naming conventions viewURL & viewData can we have same name for properties. i.e. viewurl at all places.